sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -931,26 +913,18 @@ void setUncaughtExceptionHandler(UncaughtExceptionHandler handler) {
     }
 
     @VisibleForTesting
-    @Nullable
     ErrorReport buildErrorReport(ManagedErrorLog log) {
         UUID id = log.getId();
         if (mErrorReportCache.containsKey(id)) {
             ErrorReport report = mErrorReportCache.get(id).report;
             report.setDevice(log.getDevice());
             return report;
         } else {
-            File file = ErrorLogHelper.getStoredThrowableFile(id);
-            if (file != null) {
-                String stackTrace = null;
-                if (file.length() > 0) {
-                    stackTrace = FileManager.read(file);
-                }
-                ErrorReport report = ErrorLogHelper.getErrorReportFromErrorLog(log, stackTrace);
-                mErrorReportCache.put(id, new ErrorLogReport(log, report));
-                return report;
-            }
+            String stackTrace = log.getException() == null ? null : log.getException().getStackTrace();
+            ErrorReport report = ErrorLogHelper.getErrorReportFromErrorLog(log, stackTrace);
```

Comment:
@AnastasiaKubova you can probably remove stacktrace from log.getException() (regarding https://github.com/microsoft/appcenter-sdk-android/pull/1480/files#r535530792).

Reply:
I checked remove the stacktrace here but looks like it will not work in this case, because if after restarting the error reports are not sent, then the error data will be lost because error reports are overwritten.

Reply:
You should at least be able to remove the stack trace before the `Channel.enqueue` call.

Reply:
I see latest changes solve this issue. I was wondering what kind of tests cover the new code with the stack trace building.

Reply:
Mainly this case is covered these two [tests]( https://github.com/microsoft/appcenter-sdk-android/pull/1480/files#diff-d1cf30587c3b0216759ecfd672e71cab0560bdc19d6a10894cca332ee0f2e193R1526).
But some other tests were also modified for checking it.